src/hooks/useArrayField.js

```diff
@@ -1,12 +1,11 @@
 import React, { useCallback } from 'react'
 import useForm from './useForm'
 import get from '../util/get'
+import deepEqual from '../util/deepEqual'
 
-const lengthChanged = name => ({ previous, current }) => {
+const valueChanged = name => ({ previous, current }) => {
   if (previous.values === current.values) return false
-  const currValue = (get(current.values, name) || []).length
-  const prevValue = (get(previous.values, name) || []).length
-  return currValue !== prevValue
+  return !deepEqual(previous.values[name], current.values[name])
```

Comment:
I believe to get the previous and current values, we need to use `get` to allow support for more complex data hierarchies.

Reply:
Good point.